api/src/main/java/com/basistech/rosette/api/RosetteAPI.java

```diff
@@ -1753,6 +1758,12 @@ private String headerValueOrNull(Header header) {
             throws IOException, RosetteAPIException {
         int status = httpResponse.getStatusLine().getStatusCode();
         String encoding = headerValueOrNull(httpResponse.getFirstHeader(HttpHeaders.CONTENT_ENCODING));
+        try {
+            int concurrencyHeader = Integer.parseInt(headerValueOrNull(httpResponse.getFirstHeader("X-RosetteApi-Concurrency")));
+            maxSockets = concurrencyHeader;
+        } catch (NumberFormatException e) {
+            throw e;
```

Comment:
Is the server now required to set X-RosetteApi-Concurrency?
We should do something different with the exception than just throwing it back.